### PR TITLE
Track free plan conversions in Google Analytics and AdWords.

### PR DESCRIPTION
**Pull request by jaswrks**
work in progress...

### CODE REVIEW

**Comment by MicBosi** [comment]
Awesome work @jaswrks, thanks for taking the time to create this alternative version!
I have a few questions:
- You mentioned this is a "watered-down" version of the old PR, do you mean this version supports less features/cases than the other one or simply that it requires less code changes?
- Have you tried simply removing this `if` to allow the free case to go though?
https://github.com/Automattic/wp-calypso/blob/7daab9337a83407c99ee5ede85ed0925c4c8bb1d/client/my-sites/checkout/checkout/transaction-steps-mixin.jsx#L109-L114
looks like we do get a receipt id and also have already a `hasFreeTrial()` method to distinguish the free vs non-free cases (I think the "trial" part it's a misnomer). It seems you may not even need to synthesise the cart?
- I went through the P2 post again and I also agree the `calypso_signup_complete` at the moment would the place that makes most sense to put those extra params but it's not really essential for this PR and I'm not sure people would actually use it since we've been repeatedly told to use other events (`calypso_user_registration_complete`, `wpcom_new_purchase_complete`, `wpcom_manual_renewal_complete`) when creating funnels and tracking things. I also have a few other questions on why we should be using the other events instead of the `calypso_signup_complete` event + props. The marketing data did some research on it and found discrepancies which I think also implies some of our marketing pixels tracking might be off one way or another. Also, it might be worth investigating the idea of introducing a new `calypso_conversion_complete` event that suits our exact needs with all the props. I wouldn't want us to add props to `calypso_signup_complete` and have someone start using them and then we change our mind and find a better alternative solution. Also, we may want to look into including other props like `is_new_purchase` and `is_manual_renewal` which may dictate which event we'll end up using and from where we're firing it.
Long story short I think is worth going a bit deeper in the event conversation but to avoid have that hang up this PR and leave it for a separate one. If we decide to go with the current implementation it should be a simple and easy one anyway. What do you think?
- I have some reservations on the `has_cart_items` prop, we assume it means "free-plan" but that might change in the future and we might have a cart item that says "free-plan" or something else that is free anyway. Maybe something like `is_free` to mean the user didn't have to pay anything for the order? So we fix the semantics of it but we can keep updating the underlying implementation as things change in the future. Again this might be a conversation best kept for a separate PR.
